src/batch/src/executor2/generate_series.rs

```diff
@@ -12,49 +12,96 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::sync::Arc;
-
 use futures_async_stream::try_stream;
-use risingwave_common::array::column::Column;
-use risingwave_common::array::{ArrayBuilder, DataChunk, I32ArrayBuilder};
+use risingwave_common::array::DataChunk;
 use risingwave_common::catalog::{Field, Schema};
-use risingwave_common::error::{Result, RwError};
-use risingwave_common::types::DataType;
-use risingwave_common::util::chunk_coalesce::DEFAULT_CHUNK_BUFFER_SIZE;
+use risingwave_common::error::{ErrorCode, Result, RwError};
+use risingwave_common::types::{DataType, ScalarImpl};
+use risingwave_expr::expr::build_from_prost;
 use risingwave_pb::batch_plan::plan_node::NodeBody;
 
 use crate::executor::ExecutorBuilder;
-use crate::executor2::{BoxedDataChunkStream, BoxedExecutor2, BoxedExecutor2Builder, Executor2};
-
-pub struct GenerateSeriesI32Executor2 {
-    start: i32,
-    stop: i32,
-    step: i32,
-    cur: i32, // Current value in the series.
+use crate::executor2::{
+    BoxedDataChunkStream, BoxedExecutor2, BoxedExecutor2Builder, Executor2,
+    GenerateSeriesI32Executor2, GenerateSeriesTimestampExecutor2,
+};
 
+pub struct GenerateSeriesExecutor2Wrapper {
+    input: BoxedExecutor2,
```

Comment:
Is this wrapper an intermediate state during refactor? I am thinking that they may be able to share a common executor with 2 generic types (one for start/stop/return type, another for step type).

Reply:
> Is this wrapper an intermediate state during refactor? I am thinking that they may be able to share a common executor with 2 generic types (one for start/stop/return type, another for step type).

I refer [this](https://github.com/singularity-data/risingwave/blob/96b442a96055eb9b3fbfdd86d9af199f39dd01b0/src/stream/src/executor/simple.rs#L39) to wrap two executors in one wrapper. And I don't think it's possible to share an executor using generic type, because the NativeDataTimeWrapper's binary operations don't rely on -/+ but add/sub

Reply:
> What about returning `GenerateSeriesTimestampExecutor2` or `GenerateSeriesI32Executor2` directly from the builder? Seems they do not need to go thru the wrapping executor.

that my original idea, But I found the builder logic is writen in[ macro ](https://github.com/singularity-data/risingwave/blob/main/src/batch/src/executor/mod.rs):sweat:

Reply:
It's still possible but looks hacky, by using an empty GenerateSeriesExecutor2Wrapper that actually builds one of them as `Result<BoxedExecutor2>`. Let's keep these improvements a separate issue and get it working first.